Zend/zend_attributes.c

```diff
@@ -70,6 +114,74 @@ ZEND_API zend_attribute *zend_get_parameter_attribute_str(HashTable *attributes,
 	return get_attribute_str(attributes, str, len, offset + 1);
 }
 
+ZEND_API int zend_get_attribute_value(zval *ret, zend_attribute *attr, uint32_t i, zend_class_entry *scope)
+{
+	if (i >= attr->argc) {
+		return FAILURE;
+	}
+
+	ZVAL_COPY_OR_DUP(ret, &attr->argv[i]);
+
+	if (Z_TYPE_P(ret) == IS_CONSTANT_AST) {
+		if (SUCCESS != zval_update_constant_ex(ret, scope)) {
+			zval_ptr_dtor(ret);
+			return FAILURE;
+		}
+	}
+
+	return SUCCESS;
+}
+
+static const char *target_names[] = {
+	"CLASS",
+	"FUNCTION",
+	"METHOD",
+	"PROPERTY",
+	"CLASS CONSTANT",
```

Comment:
```suggestion
	"CLASS_CONSTANT",
```

Or alternatively, and that would be my preference, make these `class`, `function` etc. I think for the error message it is more useful to have a readable entity name than to preserve the spelling of the TARGET constant.